signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt

```diff
@@ -3,15 +3,35 @@ package com.willowtreeapps.signinwithapplebutton
 import android.net.Uri
 import android.os.Parcel
 import android.os.Parcelable
+import android.view.View
+import androidx.annotation.IdRes
 import androidx.fragment.app.FragmentManager
+import com.willowtreeapps.signinwithapplebutton.view.SignInWebViewDialogFragment
 import java.util.*
 
 class SignInWithAppleService(
-    private val clientId: String,
-    private val redirectUri: String,
-    private val scope: String
+    @IdRes private val id: Int,
+    private val fragmentManager: FragmentManager,
+    private val config: SignInWithAppleConfig,
+    private val callback: (SignInWithAppleResult) -> Unit
 ) {
```

Comment:
In addition to the extension method I'd like the consumer to be able to bring up the interface anytime by using the service object directly. I imagine that's why you left this class public! But in that case I don't want the consumer to have to necessarily have a corresponding view, nor guess at what the ID is going to be used for. So the `id` property is a sore thumb here.

Let's replace `id` with an explicit `fragmentTag: String`. The view extension method can build the fragment tag using the code from the computed property, which we can remove:

```
    private val fragmentTag: String
         get() = "SignInWithAppleButton-$id-SignInWebViewDialogFragment"
```

And then in the extension method:

```
    val fragmentTag = "View-$id-SignInWebViewDialogFragment"
    val service = SignInWithAppleService(fragmentManager, fragmentTag, config, callback)
    setOnClickListener { service.show() }
```

I think this is better because if you're supplying a fragment manager, it's obvious why you're supplying a fragment tag, and you can be responsible for uniquing tags for different services. This is the same level of responsibility you'd have if you supplied an ID by hand, only, you don't have to have a view to get the ID from.

Reply:
After this change, please give some thought to this:

Does the view extension method really add anything if the service is easy to set up and invoke? Maybe we should move its functionality back into the button class.

Reply:
I think that's fair